torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -664,6 +664,14 @@ class ShapePropagator {
         }
         return;
       }
+      case prim::grad: {
+        auto tt = node->input()->type()->expect<TensorType>();
+        // grad may be undefined
+        // requires_grad may be required
+        auto grad_type = tt->withRequiresGrad({})->withPossiblyUndefined();
```

Comment:
I think the output should just be `TensorType::get()-> withPossiblyUndefined`. Grad of a tensor doesn't have any correspondence with the input tensor type.

Reply:
yes, that sounds good. i think `grad` usually keep the same dimensions as input tensors but with the data type is a bit murkier. let's make it `tensortype::get()`